### ISSUE TITLE
editor.codeActionsOnSave not triggered from autosave

### ISSUE THREAD

**Issue by richardsimko** [microsoft/vscode#123875]
Does this issue occur when all extensions are disabled?: N/A I can't verify with the eslint extension disabled, if there are other actions I can run to verify please let me know.

- VS Code Version:  1.56.0
- OS Version: macOS 11.3

Steps to Reproduce:

1. Ensure the following settings are set in `settings.json`:
```json
"files.autoSave": "onWindowChange",
"editor.codeActionsOnSave": {
      "source.fixAll.eslint": true
    },
```
2. Save file manually -> Eslint will fix errors
3. Unfocus window with unsaved files -> Files are saved but eslint is not run

This started happening with the latest Vscode update.

**Comment by FelipeGrijo** [comment]
Same on Windows with `onFocusChange`.
Windows: 10.0.19042.985
VS Code: 1.56.2

**Comment by dbaeumer** [comment]
To my knowledge the code action on save hook is only triggered by VS Code if the user manually presses save. I debugged it and the corresponding provider is not trigger on focus or window change.

Assigning to @mjbvz for final comment.

**Comment by richardsimko** [comment]
This was not the case before 1.56.0 though. This setup worked before I last upgraded and the linter fix would run even on window focus change.

**Comment by dbaeumer** [comment]
That sounds like a bug then.

**Comment by richardsimko** [comment]
That the previous behavior was a bug or the current one is? As a user it doesn't make any sense that an action called `onSave` only triggers for certain types of saves so in my opinion the current behavior is a bug.

**Comment by richardsimko** [comment]
It's probably relevant to add as well that I have Auto Save set to `onWindowChange`, just so there's no confusion here. I would only expect the linter to run as a result of the save (Which in turn is triggered by the window change due to that setting), not from the window change itself.

**Comment by mjbvz** [comment]
The previous behavior of running code actions when windows change was a bug, see #121155

While a few users like yourself wanted linter error to be fixed on window changes, many more users were confused when code was "magically" changed when they alt-tabbed away from VS Code

**Comment by richardsimko** [comment]
Very strange to have an option called `onSave` which is actually `onManualSave`. Any plans on introducing an option which is an actual `onSave` action that triggers from any type of save?

**Comment by acknudson** [comment]
I am also running into this new "fixed" functionality, and it's disrupting my workflow. I rely on the editor to run eslint and autofix my code when the file loses focus. I can understand the feedback that other users were confused, but could we not make this a configurable setting?

**Comment by richardsimko** [comment]
@dbaeumer @mjbvz Any chance you will reconsider or at the very least add a way to trigger actions on all save types? Considering there have been 3 open issues about this so far and a number of 👍  here there seems to be some demand for it.

**Comment by richardsimko** [comment]
Since there seems to be no interest from the VS Code team to fix this I've resorted to monkey patching the relevant file.

The file in question is `workbench.desktop.main.js` which on Mac is located under `/VS Code.app/Resources/app/out/vs/workbench`. What you're looking for is [these](https://github.com/microsoft/vscode/blame/23863841df396261d9234527eb467962a7dc7065/src/vs/workbench/contrib/codeEditor/browser/saveParticipants.ts#L277) minified lines. The text `createCodeActionsOnSave` doesn't seem to be touched by the minifier so I managed to find it using that and then simply removing the condition which checks for the save type. It's not an elegant solution, and of course needs to be re-applied each time VS Code is updated but at least it works.
